File: app/src/slowdebug/java/com/stedi/randomimagegenerator/app/model/data/SlowGenerator.java
package com.stedi.randomimagegenerator.app.model.data;

import android.graphics.Bitmap;
import android.support.annotation.NonNull;

import com.stedi.randomimagegenerator.ImageParams;
import com.stedi.randomimagegenerator.app.other.Utils;
import com.stedi.randomimagegenerator.generators.Generator;

public class SlowGenerator implements Generator {
    private final Generator target;

    public SlowGenerator(@NonNull Generator target) {
        this.target = target;
    }

    @Override
    public Bitmap generate(ImageParams imageParams) throws Exception {
        Utils.sleep(1000);
        return target.generate(imageParams);
    }
}

File: app/src/slowdebug/java/com/stedi/randomimagegenerator/app/model/data/SlowGenerator.kt
package com.stedi.randomimagegenerator.app.model.data

import android.graphics.Bitmap
import com.stedi.randomimagegenerator.ImageParams
import com.stedi.randomimagegenerator.app.GENERATOR_GENERATE_SLEEP
import com.stedi.randomimagegenerator.app.other.sleep
import com.stedi.randomimagegenerator.generators.Generator

class SlowGenerator(private val target: Generator) : Generator {

    override fun generate(imageParams: ImageParams): Bitmap? {
        sleep(GENERATOR_GENERATE_SLEEP)
        return target.generate(imageParams)
    }
}